README.md

```diff
@@ -111,9 +128,16 @@ When some options are not specified and the command is run with `--no-interactio
 ![Some options specified](docs/assets/images/no-interaction-error.png)
 
 
-## Limitations
+## Option configuration
+
+The following configuration options are available per option:
 
-This has only been tested with string options set to `InputOption::VALUE_REQUIRED` - any other options are (currently) unsupported.
+| Configuration value | Description | Allowed values |
+|---------------------|-------------|----------------|
+| type | The type of value to be expected (input will be transformed) | `int`, `integer`, `float`, `bool`, `boolean`, `string` (written as strings)|
+| description | The description for the option/the text to be displayed when asking for the option | _anything_ |
+| required | Whether or not the option is required. If it's set to true, then a value MUST be entered and they will be prompted until one is provided. If set to false, a default value will be used if set, or `null` will be returned as their input. | `true` or `false` (as boolean)
```

Comment:
Yeah, I added it as more of "some sort of input is required for the code to function", for example if somebody forgot to provide an `age` but a value is necessary for the command to complete, you would set it to to be required, but you could also set it to false (which is the default) if it's an entirely optional thing code-wise.